pyfr/solvers/base/elements.py

```diff
@@ -88,8 +91,26 @@ def set_ics_from_cfg(self):
             raise ValueError('Invalid constants (x, y, or z) in config file')
 
         # Get the physical location of each solution point
-        coords = self.ploc_at_np('upts').swapaxes(0, 1)
-        vars |= dict(zip('xyz', coords))
+        coords = self.ploc_at_np('upts').swapaxes(0,1)
+
+        # See if performing L2 projection
+        ename = self.basis.name
+        upts = self.cfg.get(f'solver-elements-{ename}', 'soln-pts')
+        qdeg = (self.cfg.getint('soln-ics', f'quad-deg-{ename}', 0) or 
+                self.cfg.getint('soln-ics', f'quad-deg', 0))
+        # Default to solution points if quad-pts are not specified
+        qpts = (self.cfg.get('soln-ics', f'quad-pts-{ename}', upts) or 
+                self.cfg.get('soln-ics', f'quad-pts', upts))
+        # Get prolongation/projection operators 
+        if qdeg:
+            qrule = get_quadrule(ename, qpts, qdeg=qdeg)
+            m7 = self.basis.ubasis.nodal_basis_at(qrule.pts)
```

Comment:
Not sure this is needed. `ploc_at_np` will take a list of points in lieu of a name.